src/components/text-editor/text-editor.test.js

```diff
@@ -0,0 +1,359 @@
+          innerTextList("ul", positionOfElement("third")).should(
+            "have.text",
+            "is"
+          );
+          innerTextList("ul", positionOfElement("fourth")).should(
+            "have.text",
+            "awesome"
+          );
+        } else if (buttonType === "bullet-list-numbers") {
+          innerTextList("ol", positionOfElement("second")).should(
+            "have.text",
+            "Testing"
+          );
+          innerTextList("ol", positionOfElement("third")).should(
+            "have.text",
+            "is"
+          );
+          innerTextList("ol", positionOfElement("fourth")).should(
+            "have.text",
+            "awesome"
+          );
+        }
+        textEditorToolbar(buttonType).should(
+          "have.css",
+          "background-color",
+          "rgb(204, 214, 219)"
+        );
+      }
+    );
+
+    it("should focus bold button using Tab keyboard", () => {
+      CypressMountWithProviders(<TextEditorCustom />);
+      textEditorInput().focus();
+      cy.focused().tab();
+      textEditorToolbar("bold").should("be.focused");
+    });
+
+    it.each([
+      ["bold", 0],
+      ["italic", 1],
+      ["bullet-list", 2],
+      ["number-list", 3],
+    ])(
+      "should focus %s button using RightArrow keyboard key",
+      (buttonType, times) => {
+        CypressMountWithProviders(<TextEditorCustom />);
+        textEditorInput().focus().tab();
+        for (let i = 0; i < times; i++) {
+          cy.focused().trigger("keydown", keyCode("rightarrow"));
+        }
+        textEditorToolbar(buttonType).should("be.focused");
+      }
+    );
+
+    it.each([
+      ["bold", 4],
+      ["italic", 3],
+      ["bullet-list", 2],
+      ["number-list", 1],
+    ])(
+      "should focus %s button using LeftArrow keyboard key",
+      (buttonType, times) => {
+        CypressMountWithProviders(<TextEditorCustom />);
+        textEditorInput().focus().tab();
+        for (let i = 0; i < times; i++) {
+          cy.focused().trigger("keydown", keyCode("leftarrow"));
+        }
+        textEditorToolbar(buttonType).should("be.focused");
+      }
+    );
+
+    it.each([
+      ["bold", 0],
+      ["italic", 1],
+      ["bullet-list", 2],
+      ["number-list", 3],
+    ])(
+      "should activate %s button using Enter keyboard key",
+      (buttonType, times) => {
+        CypressMountWithProviders(<TextEditorCustom />);
+        textEditorInput().focus().tab();
+        for (let i = 0; i < times; i++) {
+          cy.focused().trigger("keydown", keyCode("rightarrow"));
+        }
+        cy.focused().trigger("keydown", keyCode("Enter"));
+        textEditorToolbar(buttonType).should(
+          "have.css",
+          "background-color",
+          "rgb(204, 214, 219)"
+        );
+      }
+    );
+
+    it.each([
+      ["bold", 0],
+      ["italic", 1],
+      ["bullet-list", 2],
+      ["number-list", 3],
+    ])(
+      "should activate %s button using Enter keyboard key",
```

Comment:
"should activate %s button using Space keyboard key",

Reply:
fixed